chatkit-core/src/test/kotlin/com/pusher/chatkit/users/UserSubscriptionDispatcherTest.kt

```diff
@@ -0,0 +1,157 @@
+package com.pusher.chatkit.users
+
+import com.pusher.chatkit.rooms.api.JoinedRoomApiType
+import com.pusher.chatkit.rooms.api.JoinedRoomApiTypeMapper
+import com.pusher.chatkit.rooms.api.RoomMembershipApiType
+import com.pusher.chatkit.rooms.api.RoomReadStateApiType
+import com.pusher.chatkit.rooms.state.JoinedRoomsStateDiffer
+import com.pusher.chatkit.state.JoinedRoom
+import com.pusher.chatkit.state.JoinedRoomsReceived
+import com.pusher.chatkit.state.LeftRoom
+import com.pusher.chatkit.state.RoomDeleted
+import com.pusher.chatkit.state.RoomUpdated
+import com.pusher.chatkit.users.api.UserApiType
+import com.pusher.chatkit.users.api.UserSubscriptionDispatcher
+import com.pusher.chatkit.users.api.UserSubscriptionEvent
+import com.pusher.chatkit.util.DateApiTypeMapper
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.reduxkotlin.Dispatcher
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object UserSubscriptionDispatcherTest : Spek({
+
+    val simpleJoinedRoomApiType = JoinedRoomApiType(
+        id = "id1",
+        name = "room1",
+        createdById = "person1",
+        pushNotificationTitleOverride = "notification override",
+        private = false,
+        customData = mapOf("highlight" to "blue"),
+        lastMessageAt = "2020-02-27T17:12:10Z",
+        updatedAt = "2020-02-27T17:12:20Z",
+        createdAt = "2020-02-27T17:12:30Z",
+        deletedAt = null
+    )
+
+    val simpleUser = UserApiType(
+        id = "user1",
+        createdAt = "",
+        updatedAt = "",
+        name = "name",
+        avatarURL = null,
+        customData = null,
+        online = false)
+
+    describe("given no existing state") {
+        val dispatcher = mockk<Dispatcher>(relaxed = true)
+        val differ = mockk<JoinedRoomsStateDiffer> {
+            every { stateExists() } returns false
+        }
+        val dateApiTypeMapper = DateApiTypeMapper()
+        val joinedRoomApiTypeMapper = JoinedRoomApiTypeMapper(dateApiTypeMapper)
+        val userSubscriptionDispatcher = UserSubscriptionDispatcher(
+            joinedRoomApiTypeMapper = joinedRoomApiTypeMapper,
+            joinedRoomsStateDiffer = differ,
+            dispatcher = dispatcher
+        )
+
+        describe("when I receive an InitialState event") {
```

Comment:
```suggestion
        describe("when InitialState is received") {
```
consistency with existing concise BDD descriptions and earlier feedback on another test in this PR

Reply:
applies to other tests in this file / potentially PR

Reply:
👍 have updated all the when's in the UserSubscriptionDispatcherTest.kt - will also review the other tests.